Wikipedia/Code/FindAndReplaceKeyboardBar.swift

```diff
@@ -0,0 +1,345 @@
+        closeButton.tintColor = theme.colors.secondaryText
+        previousButton.tintColor = theme.colors.secondaryText
+        nextButton.tintColor = theme.colors.secondaryText
+        magnifyImageView.tintColor = theme.colors.secondaryText
+        findClearButton.tintColor = theme.colors.secondaryText
+        currentMatchLabel.textColor = theme.colors.tertiaryText
+        
+        replaceTextField.keyboardAppearance = theme.keyboardAppearance
+        replaceTextField.textColor = theme.colors.primaryText
+        replaceTextFieldContainer.backgroundColor = theme.colors.keyboardBarSearchFieldBackground
+        replaceButton.tintColor = theme.colors.secondaryText
+        replaceSwitchButton.tintColor = theme.colors.secondaryText
+        pencilImageView.tintColor = theme.colors.secondaryText
+        replaceClearButton.tintColor = theme.colors.secondaryText
+        replaceTypeLabel.textColor = theme.colors.tertiaryText
+        replacePlaceholderLabel.textColor = theme.colors.tertiaryText
+        
+    }
+}
+
+//MARK: Private
+
+private extension FindAndReplaceKeyboardBar {
+    
+    func hideUndoRedoIcons() {
+        if findTextField.responds(to: #selector(getter: inputAssistantItem)) {
+            findTextField.inputAssistantItem.leadingBarButtonGroups = []
+            findTextField.inputAssistantItem.trailingBarButtonGroups = []
+        }
+        
+        if replaceTextField.responds(to: #selector(getter: inputAssistantItem)) {
+            replaceTextField.inputAssistantItem.leadingBarButtonGroups = []
+            replaceTextField.inputAssistantItem.trailingBarButtonGroups = []
+        }
+    }
+    
+    func resetReplace() {
+        replaceTextField.text = nil
+        replaceType = .replaceSingle
+    }
+    
+    func updateMatchCountLabel(index: Int, total: UInt) {
+        
+        guard let findText = findTextField.text,
+            findText.count > 0 else {
+                currentMatchLabel.text = nil
+                return
+        }
+        
+        var labelText: String
+        if total > 0 && index == -1 {
+            labelText = String.localizedStringWithFormat("%lu", total)
+        } else {
+            let format = WMFLocalizedStringWithDefaultValue("find-in-page-number-matches", nil, nil, "%1$@ / %2$@", "Displayed to indicate how many matches were found even if no matches. Separator can be customized depending on the language. %1$@ is replaced with the numerator, %2$@ is replaced with the denominator.")
+            labelText = String.localizedStringWithFormat(format, NSNumber(value: index + 1), NSNumber(value: total))
+        }
+        
+        currentMatchLabel.text = labelText
+    }
+    
+    func updatePreviousNextButtonsState(total: UInt) {
+        previousButton.isEnabled = total >= 2
+        nextButton.isEnabled = total >= 2
+    }
+    
+    func updateReplaceLabelState() {
+        #warning("todo: localize")
```

Comment:
Could this (and the other `#warning`s) be updated to comments so that we keep the project at zero warnings? It's easier to see if new compiler warnings are added while developing that way

Reply:
👍 done!